pkg/controllers/node/node_controller.go

```diff
@@ -16,111 +16,68 @@ package node
 	"k8s.io/client-go/kubernetes"
-	corecache "k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
 
-	calicocache "github.com/projectcalico/kube-controllers/pkg/cache"
 	"github.com/projectcalico/kube-controllers/pkg/controllers/controller"
 	api "github.com/projectcalico/libcalico-go/lib/apis/v3"
 	client "github.com/projectcalico/libcalico-go/lib/clientv3"
 	"github.com/projectcalico/libcalico-go/lib/errors"
 	"github.com/projectcalico/libcalico-go/lib/options"
 )
 
-// This cache maps a kubernetesNodeName to its corresponding calicoNode.
-type cache struct {
-	sync.RWMutex
-	nodes map[string]string
-}
+const (
+	RateLimitCalicoList   = "calico-list"
+	RateLimitK8s          = "k8s"
+	RateLimitCalicoDelete = "calico-delete"
+)
 
 // NodeController implements the Controller interface.  It is responsible for monitoring
 // kubernetes nodes and responding to delete events by removing them from the Calico datastore.
-// It keeps a cache of known calico nodes and their corresponding kubernetes nodes to
-// accomplish this.
 type NodeController struct {
-	ctx              context.Context
-	informer         corecache.Controller
-	k8sResourceCache calicocache.ResourceCache
-	nodeLookupCache  *cache
-	calicoClient     client.Interface
-	k8sClientset     *kubernetes.Clientset
-}
-
-type nodeData struct {
+	ctx          context.Context
+	informer     cache.Controller
+	calicoClient client.Interface
+	k8sClientset *kubernetes.Clientset
+	rl           workqueue.RateLimiter
+	schedule     chan interface{}
 }
 
 // NewNodeController Constructor for NodeController
 func NewNodeController(ctx context.Context, k8sClientset *kubernetes.Clientset, calicoClient client.Interface) controller.Controller {
-	cacheArgs := calicocache.ResourceCacheArgs{
-		ObjectType: reflect.TypeOf(nodeData{}),
-		ListFunc: func() (map[string]interface{}, error) {
-			// Get all nodes from the Calico datastore
-			calicoNodes, err := calicoClient.Nodes().List(ctx, options.ListOptions{})
-			if err != nil {
-				return nil, err
-			}
-
-			// Iterate through and store the k8s nodes in our cache.
-			m := make(map[string]interface{})
-			for _, calicoNode := range calicoNodes.Items {
-				// find its kubernetes orchRef
-				if k8sNodeName := getK8sNodeName(calicoNode); k8sNodeName != "" {
-					m[k8sNodeName] = nodeData{}
-				}
-			}
-
-			log.Debugf("Found %d nodes in Calico datastore:", len(m))
-			return m, nil
-		},
-		ReconcilerConfig: calicocache.ReconcilerConfig{
-			DisableMissingInDatastore: true,
-			DisableMissingInCache:     false,
-			DisableUpdateOnChange:     false,
-		},
-	}
-
-	k8sResourceCache := calicocache.NewResourceCache(cacheArgs)
-	nodeLookupCache := cache{nodes: make(map[string]string)}
+	// channel used to kick the controller into scheduling a sync. It has length
+	// 1 so that we coalesce multiple kicks while a sync is happening down to
+	// just one additional sync.
+	schedule := make(chan interface{}, 1)
```

Comment:
nit: i believe it's more usual to use a `chan struct{}` for this kind of thing where you don't actually care about the value, but I really don't think it makes a lot of difference.